src/app/account/page.tsx

```diff
@@ -203,20 +205,26 @@ function TableTabs({
   addressToDelegatee?: AddressTo<Delegatee>;
   activateStake: (g: Address) => void;
 }) {
-  const [tab, setTab] = useState<'stakes' | 'rewards' | 'delegations'>('stakes');
+  const router = useRouter();
+  const params = useSearchParams();
+  const tabs = ['stakes', 'rewards', 'delegations', 'history'] as const;
+  const initialTab = (params.get('tab') as (typeof tabs)[number]) || tabs[0];
+  const [tab, setTab] = useState<(typeof tabs)[number]>(initialTab);
+
+  useEffect(() => {
+    if (params.get('tab') !== tab) {
+      router.push('?tab=' + tab);
+    }
+  }, [params, router, tab]);
```

Comment:
how about moving out into a separate hook?

Reply:
1. its just nice to group it together as such

2. (more involved) but we have other tab uis that we could eventually have use the same generified hook to control too

Reply:
done!